File: src/main/java/com/uitnetwork/alexa/service/AlexaIntentRequestService.java
package com.uitnetwork.alexa.service;

import org.springframework.core.annotation.Order;

import com.amazon.speech.speechlet.SpeechletResponse;
import com.uitnetwork.alexa.model.AlexaIntentRequest;

@Order(1)
public abstract class AlexaIntentRequestService {

    public abstract SpeechletResponse doProcess(AlexaIntentRequest alexaIntentRequest);

    public abstract boolean canProcess(AlexaIntentRequest alexaIntentRequest);
}

File: src/main/kotlin/com/uitnetwork/alexa/service/AlexaIntentRequestService.kt
package com.uitnetwork.alexa.service

import com.amazon.speech.speechlet.SpeechletResponse
import com.uitnetwork.alexa.model.AlexaIntentRequest
import org.springframework.core.annotation.Order

@Order(1)
abstract class AlexaIntentRequestService {

    abstract fun doProcess(alexaIntentRequest: AlexaIntentRequest): SpeechletResponse

    abstract fun canProcess(alexaIntentRequest: AlexaIntentRequest): Boolean
}
